exceptions: few examples of exceptions

dskrumov opened the pull request:
Added few samples which are illustrating how exceptions are used in Java.

dskrumov commented:
Already removed the empty spaces and changed the array excercise. 

Please make a review @mgenov .

mgenov commented on src/com/company/readnumbers/Read.java:
```diff
@@ -0,0 +1,16 @@
+package com.company.readnumbers;
+
+public class Read {
+    public void ReadNumbers(int a, int b) throws Customexception {
```
methods in java should start with lower case character. 

Also method name looks misleading as this method doesn't read anything.

mgenov commented on src/com/company/readnumbers/Read.java:
```diff
@@ -0,0 +1,16 @@
+    public void ReadNumbers(int a, int b) throws Customexception {
+        boolean result = true;
+        boolean result2 = false;
+        if (a > 10 || a < 0) {
```
What is magical in 10 ? 

Hardcoding of values is a bad practice.

mgenov commented on src/com/company/readnumbers/Read.java:
```diff
@@ -0,0 +1,16 @@
+package com.company.readnumbers;
+
+public class Read {
```
Read is a verb, but classes in java should be Nouns. 

Server, Car, Driver, Vehicle, TrackingObject and etc.

mgenov commented on src/com/company/readnumbers/Customexception.java:
```diff
@@ -0,0 +1,10 @@
+package com.company.readnumbers;
+
+
+public class Customexception extends Exception {
```
CustomException is not giving any information. 

Here are some exception names which you can see as examples: 
IllegalStateException, UserAlreadyExistsException, SystemTemporaryNotAvailableException and etc.

mgenov commented on src/com/company/readnumbers/Numbers.java:
```diff
@@ -0,0 +1,26 @@
+        this.end = end;
+    }
+
+    public void setNumbers(int a, int b) throws OutofIntegerRange {
```
isInRange

mgenov commented on src/com/company/readnumbers/Numbers.java:
```diff
@@ -0,0 +1,26 @@
+        } else if (b > end && start < b) {
+            throw new OutofIntegerRange("Out of range - Integer b - " + result2);
+        } else System.out.println(a);
+        System.out.println(result);
```
These system outs need to be removed from here.

mgenov commented on src/com/company/readnumbers/Numbers.java:
```diff
@@ -0,0 +1,26 @@
+    public void setNumbers(int a, int b) throws OutofIntegerRange {
+        boolean result = true;
+        boolean result2 = false;
+        if (a > end && start < a ) {
```
Single check should do the job

mgenov commented on src/com/company/readnumbers/Numbers.java:
```diff
@@ -0,0 +1,26 @@
+package com.company.readnumbers;
+
+public class Numbers {
```
Range

mgenov commented on src/com/company/readnumbers/Range.java:
```diff
@@ -0,0 +1,22 @@
+    private int end;
+
+
+    public void range(int start, int end) {
```
This should be constructor, not a method. 

```java
Range oneToTen = new Range(1, 10); 
oneToTen.contains(5); 
oneToTen.contains(11); // should throw exception
```

mgenov commented:
Squash your commits.

dskrumov commented:
commit d1f96162c2dce8b7c81ffd707e2e0adf647b3957
Merge: 0e64ff5 7d77530
Author: dskrumov <krumov_ood@abv.bg>
Date:   Wed Mar 1 10:23:47 2017 +0200

    Merge branch 'exceptions_tasks' of https://github.com/dskrumov/exceptions-dskrumov into exceptions_tasks

commit 0e64ff5e4575e920d50bc562c8c7716c1d3a1915
Author: dskrumov <krumov_ood@abv.bg>
Date:   Mon Feb 27 14:37:52 2017 +0200

    exceptions: few examples of exceptions
    
    These are the exceptions exercise with all changes.

commit 7d77530ec7d7a4c05279eb0f8613bfb57275689b
Author: dskrumov <krumov_ood@abv.bg>
Date:   Wed Mar 1 09:12:53 2017 +0200

    Changed the method to a constructor and modified the "contains" method

commit 6b23d773df6ed0263829ea332f57cc4b4bd21648
Author: dskrumov <krumov_ood@abv.bg>


This is the result of the git log command which shows me the last 3 commits

dskrumov commented:
commit d1f96162c2dce8b7c81ffd707e2e0adf647b3957
Merge: 0e64ff5 7d77530
Author: dskrumov <krumov_ood@abv.bg>
Date:   Wed Mar 1 10:23:47 2017 +0200

    Merge branch 'exceptions_tasks' of https://github.com/dskrumov/exceptions-dskrumov into exceptions_tasks

commit 0e64ff5e4575e920d50bc562c8c7716c1d3a1915
Author: dskrumov <krumov_ood@abv.bg>
Date:   Mon Feb 27 14:37:52 2017 +0200

    exceptions: few examples of exceptions
    
    These are the exceptions exercise with all changes.

commit 7d77530ec7d7a4c05279eb0f8613bfb57275689b
Author: dskrumov <krumov_ood@abv.bg>
Date:   Wed Mar 1 09:12:53 2017 +0200

    Changed the method to a constructor and modified the "contains" method

commit 6b23d773df6ed0263829ea332f57cc4b4bd21648
Author: dskrumov <krumov_ood@abv.bg>
Date:   Tue Feb 28 19:03:28 2017 +0200

    Aplied the new the changes

commit 006e961ea370c1beb609a012e6339e5d7e533ba9
Author: dskrumov <krumov_ood@abv.bg>
Date:   Tue Feb 28 18:24:58 2017 +0200

    Aplied all the changes

commit e8767e0d40bf1a03606b83aaa299507a8f806c3e
Author: dskrumov <krumov_ood@abv.bg>
Date:   Tue Feb 28 18:13:15 2017 +0200

    Renamed the classes and methods and changed the boolean exercise

commit 6a93ee6c041064d6537dc07b615c3a9c5a0b3dd8
Author: dskrumov <krumov_ood@abv.bg>
Date:   Mon Feb 27 15:10:46 2017 +0200

    Reformated the code and changed the array

commit 3dbcdbe0d1c211236c5309d0a5cec3beea9c1244
Author: dskrumov <krumov_ood@abv.bg>
Date:   Mon Feb 27 14:37:52 2017 +0200

    exceptions: few examples of exceptions
    
    Added few samples which are illustrating how exceptions are used in Java.

commit 79c3038d31e4513366f429dc3c679ddc42267a3b
Author: Miroslav Genov <mgenov@gmail.com>
Date:   Thu Feb 23 17:34:38 2017 +0200

    Create README.md
(END)

This is the result of the git log command which shows me the commits
